src/core/transform.js

```diff
@@ -24,13 +24,41 @@ import p5 from './main';
  * > <img style="max-width: 150px" src="assets/transformation-matrix.png"
  * alt="The transformation matrix used when applyMatrix is called"/>
  *
+ * <img style="max-width: 150px" src="assets/transformation-matrix-4-4.png"
+ * alt="The transformation matrix used when applyMatrix is called with 4x4 matrix"/>
+ *
+ * @method applyMatrix
+ * @param  {Array} arr an array of numbers - should be 6 or 16 length (2*3 or 4*4 matrix values)
+ * @chainable
+ */
+/**
+ * @method applyMatrix
+ * @param  {Number} a1 numbers which define the 2×3 matrix to be multiplied
+ * @param  {Number} a2 numbers which define the 2×3 matrix to be multiplied
+ * @param  {Number} a3 numbers which define the 2×3 matrix to be multiplied
+ * @param  {Number} a4 numbers which define the 2×3 matrix to be multiplied
+ * @param  {Number} a5 numbers which define the 2×3 matrix to be multiplied
+ * @param  {Number} a6 numbers which define the 2×3 matrix to be multiplied
+ * @chainable
+ */
+/**
  * @method applyMatrix
- * @param  {Number|Array} a numbers which define the 2×3 matrix to be multiplied, or an array of numbers
- * @param  {Number} b numbers which define the 2×3 matrix to be multiplied
- * @param  {Number} c numbers which define the 2×3 matrix to be multiplied
- * @param  {Number} d numbers which define the 2×3 matrix to be multiplied
- * @param  {Number} e numbers which define the 2×3 matrix to be multiplied
- * @param  {Number} f numbers which define the 2×3 matrix to be multiplied
+ * @param  {Number} b1  numbers which define the 4x4 matrix to be multiplied
+ * @param  {Number} b2  numbers which define the 4x4 matrix to be multiplied
+ * @param  {Number} b3  numbers which define the 4x4 matrix to be multiplied
+ * @param  {Number} b4  numbers which define the 4x4 matrix to be multiplied
+ * @param  {Number} b5  numbers which define the 4x4 matrix to be multiplied
+ * @param  {Number} b6  numbers which define the 4x4 matrix to be multiplied
+ * @param  {Number} b7  numbers which define the 4x4 matrix to be multiplied
+ * @param  {Number} b8  numbers which define the 4x4 matrix to be multiplied
+ * @param  {Number} b9  numbers which define the 4x4 matrix to be multiplied
+ * @param  {Number} b10  numbers which define the 4x4 matrix to be multiplied
+ * @param  {Number} b11  numbers which define the 4x4 matrix to be multiplied
+ * @param  {Number} b12  numbers which define the 4x4 matrix to be multiplied
+ * @param  {Number} b13  numbers which define the 4x4 matrix to be multiplied
+ * @param  {Number} b14  numbers which define the 4x4 matrix to be multiplied
+ * @param  {Number} b15  numbers which define the 4x4 matrix to be multipliedo
+ * @param  {Number} b16  numbers which define the 4x4 matrix to be multiplied
```

Comment:
I'm afraid this is still not quite what I mean. The original parameter names for a 2x3 matrix should be retained and reused for a 4x4 matrix with descriptions that fit both 2x3 and 4x4 at the same time. Then in the signature definition for 4x4 the name that were present in 2x3 were repeated without description and the additional parameters will have unique 4x4 only description.

Something like this:
```
/**
  * @method applyMatrix
  * @param  {Number} a numbers which define the 2×3 or 4x4 matrix to be multiplied
  * @param  {Number} b numbers which define the 2×3 or 4x4 matrix to be multiplied
  * @param  {Number} c numbers which define the 2×3 or 4x4 matrix to be multiplied
  * @param  {Number} d numbers which define the 2×3 or 4x4 matrix to be multiplied
  * @param  {Number} e numbers which define the 2×3 or 4x4 matrix to be multiplied
  * @param  {Number} f numbers which define the 2×3 or 4x4 matrix to be multiplied
  */
/**
  * @method applyMatrix
  * @param  {Number} a
  * @param  {Number} b
  * @param  {Number} c
  * @param  {Number} d
  * @param  {Number} e
  * @param  {Number} f
  * @param {Number} g numbers which define the 4x4 matrix to be multiplied
  * @param {Number} h numbers which define the 4x4 matrix to be multiplied
  * @param {Number} i numbers which define the 4x4 matrix to be multiplied
  * @param {Number} j numbers which define the 4x4 matrix to be multiplied
  * @param {Number} k numbers which define the 4x4 matrix to be multiplied
  * @param {Number} l numbers which define the 4x4 matrix to be multiplied
  * @param {Number} m numbers which define the 4x4 matrix to be multiplied
  * @param {Number} n numbers which define the 4x4 matrix to be multiplied
  * @param {Number} o numbers which define the 4x4 matrix to be multiplied
  * @param {Number} p numbers which define the 4x4 matrix to be multiplied
```

Unless there is some other reasons this could not work?

Reply:
Apologies, I mis understood :)  This has been done.  Let me know if you are now happy with the changes or if anything else is needed.